### ISSUE TITLE
Override/Revert showing up with no changes on extSdCard synced folders

### ISSUE THREAD

**Issue by Catfriend1** [Catfriend1/syncthing-android#454]
### Description of the issue
Override/Revert showing up with no changes on extSdCard synced folders

### Version Information
    App Version: 1.2.0.8
    Syncthing Version: v1.2.0
    Android Version: Android 10 @ 2019-06-05
    Device manufacturer: Google AVD 10.x
    SDCARD Filesystem: Mounted via sdcardfs.

**Comment by Catfriend1** [comment]
Related:
- https://github.com/syncthing/syncthing/pull/5852
- https://forum.syncthing.net/t/ignore-permissions-different-mtimes-on-android-sdcard/13342

**Comment by Catfriend1** [comment]
Test made with AppV 1.2.0.8, "testfolder", syncing one file from SENDONLY (PC) to SENDRECEIVE (phone):

- File synced over - up2date
- File has original mtime on phone (like pc)  ModTime:1995-10-14 17:51:38 +0000 UTC
- App restart - no change
- Phone reboot
- PC shows "override changes" button - Phone instance reports in DB the mtime is now  ModTime:1995-10-14 21:51:38 +0000 UTC

Logs
[st_sendrecv1_received_testfile.log](https://github.com/Catfriend1/syncthing-android/files/3427252/st_sendrecv1_received_testfile.log)
[st_sendrecv2_override_btn_on_other_side.log](https://github.com/Catfriend1/syncthing-android/files/3427253/st_sendrecv2_override_btn_on_other_side.log)

**Comment by Catfriend1** [comment]
Thanks for all your efforts on this, imsodin. Unfortunately I have to report back with no good news. We’re almost game over with SDcards I think…

Short summary:

I’ve synced a file (like in my previous test scenario from PR #5852) from PC to phone. The file is created on the phone with the correct “original” mtime on sdcard. I exit the app, verify the original mtime is still there and reboot the phone. I ensure to NOT start the Syncthing App and directly go to my file manager app. I see for about 10 seconds (if I’m fast getting there after bootup) the mtime is still correct and then - poof - it is the “original mtime +4 hours” magically.
I suspect this is some Android OS service doing anything to it, but I’m currently trying to find out which one it might be. Btw I ran the tests on SDCARDFS (which is used in Android 8+) - here using an emulator with the recent Android 10 image from Google.

What makes me double confused is, when I hit the “override change” button on PC and the phone is “sendreceive”, I get the original timestamp back and then - magically - it stays like this even after reboot.

**Comment by Catfriend1** [comment]
Here's a good reproducer for the "sdcardfs - Android OS modifies mtime after reboot silently" problem:
- push a test file to phone using
> adb push baseconv2.exe /storage/10F1-1918/Android/data/com.github.catfriend1.syncthingandroid.debug/files/baseconv2.exe

- Open your favorite file manager app and note the mtime (which is "original")
- Reboot phone.
- Open your favorite file manager app and note the mtime has CHANGED silently (without running Syncthing) and is now "+4 hours"

**Comment by Catfriend1** [comment]
New test scenario - without the Syncthing App:
- Disabled system app "Media storage" (Media Scanner?)
- adb push'ing test file ... original mtime is there
- Rebooted phone.
- mtime changed, very suprising is, that the mtime now changed to "now - 2 hours" instead of "original + 4 hours".

**Comment by Catfriend1** [comment]
Damn! It's even done by Android OS in the SDCARD root folder, see logs:
```
07-24 15:08:35.537 +0000  1836  1836 D vold    : Recognized experimental block major ID 253 as virtio-blk (emulator's virtual SD card device)
07-24 15:08:35.583 +0000  1836  1836 D vold    : /dev/block/vold/disk:253,80: LABEL="SDCARD" UUID="10F1-1918" TYPE="vfat"
07-24 15:08:35.595 +0000  1836  1836 D vold    : /dev/block/vold/public:253,80: LABEL="SDCARD" UUID="10F1-1918" TYPE="vfat"
07-24 15:08:35.595 +0000  1836  1836 D vold    : /system/bin/fsck_msdos
07-24 15:08:35.681 +0000  2654  2654 W sdcard  : Device explicitly enabled sdcardfs



07-24 15:05:42.145 +0000  4969  5729 I SDCardScannerService: getExternalFilesDirs /storage/emulated/0/Android/data/org.fdroid.fdroid/files
07-24 15:05:42.145 +0000  4969  5729 I SDCardScannerService: getExternalFilesDirs /storage/emulated/0/Android/data/org.fdroid.fdroid/files
07-24 15:05:42.157 +0000  4969  5729 I SDCardScannerService: getExternalFilesDirs /storage/10F1-1918/Android/data/org.fdroid.fdroid/files
07-24 15:05:42.157 +0000  4969  5729 I SDCardScannerService: getExternalFilesDirs /storage/10F1-1918/Android/data/org.fdroid.fdroid/files
07-24 15:05:42.162 +0000  4969  5729 I SDCardScannerService: sdcard files []


adb shell ls /storage/10F1-1918 -l
-rwxrwx--x 1 root sdcard_rw 104960 1995-10-14 17:51 baseconv2.exe

adb shell ls /storage/10F1-1918 -l
-rwxrwx--x 1 root sdcard_rw 104960 1995-10-14 21:51 baseconv2.exe
```

**Comment by Catfriend1** [comment]
It is indeed solved according to my most recent test on a Motorola X Play (lux) running on Android 8.0 (LineAgeOS without root).

I noticed one problem, as @rahrah said , the file system auto-detection doesn't work properly on Android 8.0 (I mean the tested device with this).

Immediately on startup, even before the "Syncthing version XX" line comes up, it logs this:

> DEBUG: Detecting FS at /storage/D9D0-1BFD/Android/media/com.github.catfriend1.syncthingandroid.debug/extsdtest on android, found , thus leaving mtime window at 0

If this is not a problem with PR #5852 's code, I suggest setting mtimeWindow to 2 secs from the wrapper. Tested this and it worked fine. I also learned that the problem only can be reproduced with a REAL device using SDCARD and syncing a file with an UNEVEN-second timestamp to it.
